### PR TITLE
feat: tune default settings to align with java-bigtable

### PR DESCRIPTION
**Pull request by rahulKQL**
Towards #2454 

This change updates bigtable-client-core default values to match with veneer counterpart. This is needed to provide a smoother transition to users. 

## Updated fields

|                      Classic configuration                     |                 Existing value                |      Aligned value      | Aligned To                                                                                                      |
|:--------------------------------------------------------------:|:---------------------------------------------:|:-----------------------:|-----------------------------------------------------------------------------------------------------------------|
| DataChannelCount                                               | CPU * 4                                       | CPU * 2                 | transportChannelProvider -> setPoolSize                                                                         |
| BulkOptions -> bulkMaxRowKeyCount                              | 125                                           | 100                     | bulk(mutate/read) -> <br>batchingSettings -> elementCountThreshold                                              |
| BulkOptions -> bulkMaxRequestSize                              | 1MB                                           | 20MB                    | bulkMutate -> batching -> requestByteSize                                                                       |
| BulkOptions -> autoFlushMs                                     | 0 (indefinitely)                              | 1 seconds               | bulkMutate -> batching -> delayThreshold                                                                        |
| BulkOptions -> maxInflightRpcs                                 | 10 * channelCount                             | 10 * 100 * channelCount | bulkMutate -> <br>flowController -> maxOutstandingElementCount                                                  |
| BulkOptions -> maxMemory                                       | 10% of available memory <br>with a max of 1GB | 100 MB                  | bulkMutate -> <br>flowController -> maxOutstandingRequestBytes                                                  |
| RetryOptions -> initialBackoffMillis                           | 5 ms                                          | 10 ms                   | RetrySettings -> initialRetryDelay                                                                              |
| RetryOptions -> maxElapsedBackoffMillis                        | 1 min                                         | 10 mins                 | RetrySettings -> totalTimeout                                                                                   |
| RetryOptions -> backoffMultiplier                              | 1.5                                           | 2.0                     | RetrySettings -> RetryDelayMultiplier                                                                           |
| RetryOptions -> ReadPartialRowTimeoutMillis                    | 1 min                                         | 5 mins                  | readRows -> <br>RetrySettings -> initialRpcTimeout                                                              |
| RetryOptions -> maxScanTimeoutRetries                          | 3                                             | 10                      | readRows -> <br>RetrySettings -> maxAttempts                                                                    |
| CallOptionsConfig -> shortRpcTimeoutMs                         | 60 seconds                                    | 20 seconds              | Idempotent op -> <br>RetrySetting -> initialRpc,maxRpcTimeout<br><br>NonIdempotent Op -> simpleTimeoutNoRetries |
| CallOptionsConfig -> <br>mutateRpcTimeoutMs/longRpcTimeoutMs   | 600 seconds                                   | 600 seconds             | mutateRow -> <br>RetrySettings -> totalTimeout                                                                  |
| CallOptionsConfig -> <br>readRowsRpcTimeoutMs/longRpcTimeoutMs | 600 seconds                                   | 12 hour                 | readRows -> <br>RetrySettings -> totalTimeout                                                                   |

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java [hunk 1]
```diff
@@ -94,10 +94,15 @@
   private static final Logger LOG = new Logger(BigtableOptions.class);
 
   private static int getDefaultDataChannelCount() {
-    // 20 Channels seemed to work well on a 4 CPU machine, and this ratio seems to scale well for
-    // higher CPU machines. Use no more than 250 Channels by default.
+    // Use no more than 250 Channels by default.
     int availableProcessors = Runtime.getRuntime().availableProcessors();
-    return Math.min(250, Math.max(1, availableProcessors * 4));
+    return Math.min(250, Math.max(1, availableProcessors * 2));
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java]: Lets set the default min to 4

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BulkOptions.java [hunk 2]
```diff
@@ -73,30 +73,29 @@
...
    *
    * <p>For internal use only - public for technical reasons.
    */
   @InternalApi("For internal usage only")
-  public static long BIGTABLE_BULK_AUTOFLUSH_MS_DEFAULT = 0;
+  public static long BIGTABLE_BULK_AUTOFLUSH_MS_DEFAULT = 1_000;
 
   /**
-   * Default rpc count per channel.
+   * Default rpc count per channel (default value: 1000).
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BulkOptions.java]: default value: 1 second

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BulkOptions.java [hunk 3]
```diff
@@ -73,30 +73,29 @@
...
    *
    * <p>For internal use only - public for technical reasons.
    */
   @InternalApi("For internal usage only")
-  public static final int BIGTABLE_MAX_INFLIGHT_RPCS_PER_CHANNEL_DEFAULT = 10;
+  public static final int BIGTABLE_MAX_INFLIGHT_RPCS_PER_CHANNEL_DEFAULT = 10 * 100;
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BulkOptions.java]: This looks wrong, I dont think we want 1000 rpcs / channel

Comment (rahulKQL) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BulkOptions.java]: I am a little confused here, We have the exact figure for veneer client:
https://github.com/googleapis/java-bigtable/pull/224/files#diff-7c807fb72c1041907eaeed70e8f43243R558-R562

Should we have a little different figure for the classic client?

Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BulkOptions.java]: I think we want 10 outstanding RPCs per channel.

1 rpc = 100 elements

so in veneer MaxOutstandingElementCount = channel_count * 100 * 10

Here, we are setting an rpc, so BIGTABLE_MAX_INFLIGHT_RPCS_PER_CHANNEL_DEFAULT = 10

Comment (rahulKQL) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BulkOptions.java]: Thanks alot for the explanation.

Have addressed feedback, PTAL

**Pull request merged**